sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -274,9 +274,7 @@ class SparkSession private(
    * @since 2.0.0
    */
   @transient
-  lazy val emptyDataFrame: DataFrame = {
-    createDataFrame(sparkContext.emptyRDD[Row].setName("empty"), StructType(Nil))
-  }
+  lazy val emptyDataFrame: DataFrame = Dataset.ofRows(self, LocalRelation())
```

Comment:
```suggestion
  lazy val emptyDataFrame: DataFrame = emptyDataset(RowEncoder(new StructType()))
```
?

Reply:
Why would that be better? This is basically re-implementing Dataset.ofRows.

Reply:
This is arguable for sure. DataFrame is Dataset[Row]. We already have a function which can construct empty Dataset[T], why not just re-use it. `emptyDataFrame` is some kind of specialization of `emptyDataset`.

Reply:
If we go this way, maybe: `emptyDataset(RowEncoder(StructType(Nil)))`.

Reply:
`Dataset.ofRows(self, LocalRelation())` looks simpler as I don't need to jump into another method when reading the code.